new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java

```diff
@@ -139,19 +357,40 @@ public static boolean createNamespace(String name) throws ApiException {
    *
    * @param name - name of namespace
    * @return true if successful delete, false otherwise
-   * @throws ApiException - if Kubernetes client API call fails
+   * @throws ApiException - missing required configuration data, if Kubernetes request fails or
+   *     unsuccessful
    */
   public static boolean deleteNamespace(String name) throws ApiException {
+    V1ObjectMeta metdata = new V1ObjectMetaBuilder().withName(name).build();
+    V1Namespace namespace = new V1NamespaceBuilder().withMetadata(metdata).build();
+    return deleteNamespace(namespace);
+  }
 
-    GenericKubernetesApi<V1Namespace, V1NamespaceList> namespaceClient =
-        new GenericKubernetesApi<>(
-                V1Namespace.class, // the api type class
-                V1NamespaceList.class, // the api list type class
-                "", // the api group
-            "v1", // the api version
-                "namespaces", // the resource plural
-                apiClient //the api client
-        );
+  /**
+   * Delete a Kubernetes namespace
+   *
+   * @param namespace - V1Namespace object containing name space configuration data
+   * @return true if successful
+   * @throws ApiException - missing required configuration data, if Kubernetes request fails or
+   *     unsuccessful
+   */
+  public static boolean deleteNamespace(V1Namespace namespace) throws ApiException {
+    if (namespace == null) {
+      throw new ApiException(
+          "Parameter 'namespace' is null when calling deleteNamespace()");
+    }
+
+    if (namespace.getMetadata() == null) {
```

Comment:
i don't think any of these should throw the k8s `ApiException` - that does not make sense - there are no k8s API calls here, you are just checking your args. 
`IllegalArgsException` would be a better choice.